hazelcast/src/main/java/com/hazelcast/map/impl/operation/GetWithProjectionOperation.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2008, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.projection.Projection;
+import com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl;
+
+import java.io.IOException;
+
+import static com.hazelcast.spi.ExecutionService.ASYNC_EXECUTOR;
+
+public final class GetWithProjectionOperation extends GetOperation {
+
+    private Projection projection;
+
+    public GetWithProjectionOperation() {
+    }
+
+    public GetWithProjectionOperation(String name, Data dataKey, Projection projection) {
+        super(name, dataKey);
+
+        this.dataKey = dataKey;
+        this.projection = projection;
+    }
+
+    @Override
+    public void run() {
+        Object value = recordStore.get(dataKey, false);
+        runOffloadedProjection(value, (OperationServiceImpl) getNodeEngine().getOperationService());
+    }
+
+    private void runOffloadedProjection(final Object value, final OperationServiceImpl ops) {
+        ops.onStartAsyncOperation(this);
+        getNodeEngine().getExecutionService().execute(ASYNC_EXECUTOR, new Runnable() {
```

Comment:
For the time being it is ok, but in the future having control on the executor is the better choice.